### ISSUE TITLE
Get a 1:1 of the original if sources brought back together

### ISSUE THREAD

**Issue by Gundz** [facebookresearch/demucs#461]
## ❓ Questions

Hello,

Is there a setting i can tweak to have a 1:1 of the original song if i put back all sources together ?

For the context: I'm using demucs to create Stems files ( https://www.stems-music.com ) which are supported in my DJ software (Traktor).

But i noticed that with all 4 sources playing it's not a 1:1 of the original song, which is obviously bad.
The original file is contained in the Stems format, but Traktor doesn't read it, only the 4 sources.

Thanks in advance.

**Comment by Dyslexicon** [comment]
Indeed, it would be ideal if the separated stems, pasted atop one another in a DAW or editor, and then exported/saved to mixdown, resulted in a file identical to the original. (assuming no alterations were made to the stems).

This may not be entirely possible especially if the input source is "brickwalled", but this is a surely goal worth pursuing.

**Comment by CarlGao4** [comment]
> But i noticed that with all 4 sources playing it's not a 1:1 of the original song, which is obviously bad.

In fact, before writing output files, all the stems are rescaled to avoid clip (or avoid part of the audio is too loud to be saved in an ordinary file). To avoid this, you can add `--float32 --clip-mode none` to your command (also remove `--mp3`). But remember that this will cause the output file become much bigger! 

P.S. I'm not an English native speaker, but I hope you can understand me

**Comment by Gundz** [comment]
> In fact, before writing output files, all the stems are rescaled to avoid clip (or avoid part of the audio is too loud to be saved in an ordinary file). To avoid this, you can add `--float32 --clip-mode none` to your command (also remove `--mp3`). But remember that this will cause the output file become much bigger!

Hello !
Thanks for the precision, but there is no "--clip-mode none" option:
`demucs.separate: error: argument --clip-mode: invalid choice: 'none' (choose from 'rescale', 'clamp')`
(tested on main and v4.0.0)

But i tried `--float32` and `--clip-mode clamp` but i have (almost) the same result as the default settings (no `--mp3` used)

**Comment by AnshulJ999** [comment]
Having this would be really great. I'm using demucs to create backing tracks for songs I'd like to learn and cover, but I if A/B the original song versus the set of stems, there's a difference in amplitude and high-frequency content. I'm guessing the frequency is because (I think) demucs doesn't process the whole 44.1kHz?

Still, having some flag to make it 1:1 the original would be amazing.
